src/wallet/wallet.cpp

```diff
@@ -5125,76 +5129,76 @@ void CWallet::ResendWalletTransactions(int64_t nBestBlockTime)
  */
 
 
-CAmount CWallet::GetBalance(const isminefilter& filter, const int min_depth) const
+CAmount CWallet::GetBalance(const std::optional<int>& asOfHeight, const isminefilter& filter, const int min_depth) const
 {
     CAmount nTotal = 0;
     {
         LOCK2(cs_main, cs_wallet);
         for (const auto& entry : mapWallet)
         {
             const CWalletTx* pcoin = &entry.second;
-            if (pcoin->IsTrusted() && pcoin->GetDepthInMainChain() >= min_depth) {
-                nTotal += pcoin->GetAvailableCredit(true, filter);
+            if (pcoin->IsTrusted(asOfHeight) && pcoin->GetDepthInMainChain(asOfHeight) >= min_depth) {
+                nTotal += pcoin->GetAvailableCredit(asOfHeight, true, filter);
             }
         }
     }
 
     return nTotal;
 }
 
-CAmount CWallet::GetUnconfirmedBalance() const
+CAmount CWallet::GetUnconfirmedBalance(const std::optional<int>& asOfHeight) const
```

Comment:
Again, I think this should mean only mined but unconfirmed balance, when `asOfHeight` is set.

Reply:
I'm not sure that it makes sense to have `asOfHeight` as an argument to `GetUnconfirmedBalance`. Actually, though... this would make sense to me *if* we were passing a confirmation depth, unless this method is actually referring to "unconfirmed" as meaning "is in the mempool and has not yet been mined". If that's the case, maybe we could rename this to something like "GetUnminedBalance" or "GetMempoolBalance"

Since this is currently checking `pcoin->GetDepthInMainChain() == 0` I think that it shouldn't take the asOfHeight.